tools/RAiDER/models/gmao.py

```diff
@@ -0,0 +1,171 @@
+        self._k1 = 0.776  # [K/Pa]
+        self._k2 = 0.233 # [K/Pa]
+        self._k3 = 3.75e3 # [K^2/Pa]
+
+        # horizontal grid spacing
+        self._lat_res = 0.25
+        self._lon_res = 0.3125
+        self._x_res = 0.3125
+        self._y_res = 0.25
+
+        self._Name = 'GMAO'
+        self._files = None
+        self._bounds = None
+
+        # Projection
+        self._proj = CRS.from_epsg(4326)
+
+    
+    
+    
+    def _fetch(self, lats, lons, time, out, Nextra = 2):
+        '''
+        Fetch weather model data from GMAO: note we only extract the lat/lon bounds for this weather model; fetching data is not needed here as we don't actually download any data using OpenDAP
+        '''
+        # bounding box plus a buffer
+        lat_min, lat_max, lon_min, lon_max = self._get_ll_bounds(lats, lons, Nextra)
+        self._bounds = (lat_min, lat_max, lon_min, lon_max)
+
+
+    
+    
+
+    def load_weather(self,f):
+        '''
+        Consistent class method to be implemented across all weather model types.
+        As a result of calling this method, all of the variables (x, y, z, p, q,
+        t, wet_refractivity, hydrostatic refractivity, e) should be fully
+        populated.
+        '''
+        self._load_model_level(f)
+
+
+    
+    def _load_model_level(self,f):
+        '''
+        Get the variables from the GMAO link using OpenDAP
+        '''
+        # calculate the array indices for slicing the GMAO variable arrays
+        lat_min_ind = int((self._bounds[0] - (-90.0)) / self._lat_res)
+        lat_max_ind = int((self._bounds[1] - (-90.0)) / self._lat_res)
+        lon_min_ind = int((self._bounds[2] - (-180.0)) / self._lon_res)
+        lon_max_ind = int((self._bounds[3] - (-180.0)) / self._lon_res)
+        
+        import datetime as dt
+        T0 = dt.datetime(2017,12,1,0,0,0)
+        DT = self._time - T0
+        time_ind = int(DT.total_seconds() / 3600.0 / 3.0)
+        
+        ml_min = 0
+        ml_max = 71
+
+        # open the dataset and pull the data
+        import pydap.client
+        import pydap.cas.urs
+        url = 'https://opendap.nccs.nasa.gov/dods/GEOS-5/fp/0.25_deg/assim/inst3_3d_asm_Nv'
+        session = pydap.cas.urs.setup_session('username', 'password', check_url=url)
+        ds = pydap.client.open_url(url, session=session)
+        
+        q = ds['qv'].array[time_ind,ml_min:(ml_max+1),lat_min_ind:(lat_max_ind+1),lon_min_ind:(lon_max_ind+1)][0]
+        p = ds['pl'].array[time_ind,ml_min:(ml_max+1),lat_min_ind:(lat_max_ind+1),lon_min_ind:(lon_max_ind+1)][0]
+        t = ds['t'].array[time_ind,ml_min:(ml_max+1),lat_min_ind:(lat_max_ind+1),lon_min_ind:(lon_max_ind+1)][0]
+        h = ds['h'].array[time_ind,ml_min:(ml_max+1),lat_min_ind:(lat_max_ind+1),lon_min_ind:(lon_max_ind+1)][0]
+        
+        # calculate the lat, lon and mean h for each layer in the regular grid
+        hs = np.zeros(72)
+        for ii in range(72):
+            hs[ii] = np.mean(h[ii,:,:])
```

Comment:
I think you can use the `axis` parameter of [np.mean](https://numpy.org/doc/stable/reference/generated/numpy.mean.html). This will be much faster, less code and doesn't have a hardcoded constant.

```suggestion
        hs = np.mean(h, axis=(1, 2))
```